### PR TITLE
CV3X-127 HA for rsyslog

### PR DESCRIPTION
**Pull request by dchirikov**
configure rsyslog standalone and with HA support

### CODE REVIEW

File: configuration/controller/rsyslog.sh [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+            echo "${CTRL2_IP}"
+            return
+        fi
+    done
+    echo "${CTRL1_IP}"
+}
+
+echo_info "Check config variables available."
+
+echo "RSYSLOG_CLIENT_NETWORK=${RSYSLOG_CLIENT_NETWORK?"Should be defined"}"
```
Comment (jflf-CV) [inline configuration/controller/rsyslog.sh]: Please don't use the `${?}` expansion pattern. Again, see documentation for explanation:
https://github.com/clustervision/trinityX/blob/development/doc/config_post_scripts.rst#shell-script-error-management

File: configuration/controller/rsyslog.sh [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License (included with the sources) for more
+# details.
+######################################################################
+
+
+set -ex
```
Comment (jflf-CV) [inline configuration/controller/rsyslog.sh]: Please don't use `set -e`. See the documentation:
https://github.com/clustervision/trinityX/blob/development/doc/config_post_scripts.rst#shell-script-error-management

File: configuration/controller/rsyslog.sh [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License (included with the sources) for more
+# details.
+######################################################################
+
+
```
Comment (jflf-CV) [inline configuration/controller/rsyslog.sh]: Where is the call to `display_var`?

File: configuration/controller/rsyslog.sh (hunk 1 above)
Comment (dchirikov) [inline configuration/controller/rsyslog.sh]: "For example, the parameter expansion ${PARAMETER:?MESSAGE} displays the message if the parameter is null or unset if the shell is interactive, but exits the shell if it's not interactive. "

It's expected. We should not run script if mandatory parameters are unset.

File: configuration/controller/rsyslog.sh (hunk 3 above)
Comment (dchirikov) [inline configuration/controller/rsyslog.sh]: Using https://github.com/clustervision/trinityX/pull/142/files/18348b32eec513f81b6e26d3f7b5ab1fc8787b80#diff-59520fec74486c9813dfa37452be2c33R358
instead.

File: configuration/controller/rsyslog.sh [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+    if [ $# -eq 2 ]; then
+        FROM=${1}
+        TO=${!FROM}
+        FILE=${2}
+    fi
+    sed -i -e "s/{{ ${FROM} }}/${TO//\//\\/}/g" $FILE
+}
+
+function get_partners_ip() {
+    if [ -z "${HA}" -o  "x${HA}" = "x0"  ]; then
```
Comment (dchirikov) [inline configuration/controller/rsyslog.sh]: Thanks fixed in 370f47a

File: configuration/controller/rsyslog.sh [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+    done
+    echo "${CTRL1_IP}"
+}
+
+display_var RSYSLOG_CLIENT_NETWORK RSYSLOG_CLIENT_NETWORK_PREFIX RSYSLOG_MESSAGES_PATH
+
+
+echo_info "Check config variables available."
+
+echo "RSYSLOG_CLIENT_NETWORK=${RSYSLOG_CLIENT_NETWORK?"Should be defined"}"
```
Comment (jflf-CV) [inline configuration/controller/rsyslog.sh]: Please do not use that syntax, it has numerous issues that are well documented. Check explicitly whether the variables have been defined or not.
https://github.com/clustervision/trinityX/blob/development/doc/config_post_scripts.rst

File: configuration/controller/rsyslog.sh [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License (included with the sources) for more
+# details.
+######################################################################
+
+
+set -e
```
Comment (jflf-CV) [inline configuration/controller/rsyslog.sh]: Please do not use ``set -e``. This is a debug feature, which is set for you when you use the ``--hardstop`` flag of the configuration tool. ``set -e`` prevents proper error management, and will fail on any command that returns a non-zero error code, including when that error code is expected. Those issues are well documented.
https://github.com/clustervision/trinityX/blob/development/doc/config_post_scripts.rst

File: configuration/controller.cfg [hunk 7]
```diff
@@ -353,3 +354,12 @@ LUNA_NETWORK_NAME=cluster
 LUNA_DHCP_RANGE_START=10.30.128.1
 LUNA_DHCP_RANGE_END=10.30.255.200
 
+
+#-----------------------------------------------------------
+# rsyslog
+#-----------------------------------------------------------
+
+RSYSLOG_MESSAGES_PATH="/var/log/cluster-messages"
+RSYSLOG_CLIENT_NETWORK=${LUNA_NETWORK}
```
Comment (jflf-CV) [inline configuration/controller.cfg]: Rsyslog can be installed on clusters that don't include Luna, so the Luna configuration might not be here. Is there any way that rsyslog can be configured only with the IPs of the controllers?

File: configuration/controller/rsyslog.sh [hunk 8]
```diff
@@ -0,0 +1,94 @@
...
+    sed -i -e "s/{{ ${FROM} }}/${TO//\//\\/}/g" $FILE
+}
+
+function get_partners_ip() {
+    if flag_is_unset HA; then
+        echo ''
+        return
+    fi
+    if [ "x${CTRL1_IP}" = "x" -o "x${CTRL2_IP}" = "x" ]; then
+        echo_error "Unable to determine IP of the partner in HA config. Exiting."
```
Comment (jflf-CV) [inline configuration/controller/rsyslog.sh]: Just use TRIX_CTRL{1,2}_IP and trust that they are set up (this happens in the critical section PS).

File: configuration/controller/rsyslog.sh (hunk 6 above)
Comment (dchirikov) [inline configuration/controller/rsyslog.sh]: JF, we discussed it, it *DOES NOT* prevent error handling.

**Pull request merged**